internal.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+# This script is used by crc developer or internal CI to build the KAO/KCMO patch images to
+# brew and then mirror built container images to quay.io/crcont. Pull secret should contain
+# the auth for `quay.io/crcont` before providing to this script.
+#    - Since this script uses rhpkg and kinit commands, it is tested on linux (not mac/windows)
+#    - As of now this script works with 4.11 releases because only `rhaos-4.11-rhel-8` branch
+#      created in dist-git and tested. 
+# Usage:
+# If you want to build latest candidate stream for 4.11
+#   - ./internal.sh 
+# If you want to build specific version of 4.11.3
+#   - OPENSHIFT_VERSION=4.11.3 ./internal.sh
+
+set -exuo pipefail
+
+export LC_ALL=C.UTF-8
+export LANG=C.UTF-8
+
+# kill all the child processes for this script when it exits
+trap 'jobs=($(jobs -p)); [ -n "${jobs-}" ] && ((${#jobs})) && kill "${jobs[@]}" || true' EXIT
+
+rm -fr crc-cluster-kube-apiserver-operator
+rm -fr crc-cluster-kube-controller-manager-operator
+
+function run_preflight_checks() {
+        if [ -z "${OPENSHIFT_PULL_SECRET_PATH-}" ]; then
+            echo "OpenShift pull secret file path must be specified through the OPENSHIFT_PULL_SECRET_PATH environment variable"
+            exit 1
+        elif [ ! -f ${OPENSHIFT_PULL_SECRET_PATH} ]; then
+            echo "Provided OPENSHIFT_PULL_SECRET_PATH (${OPENSHIFT_PULL_SECRET_PATH}) does not exists"
+            exit 1
+        fi
+}
+
+run_preflight_checks
+
+MIRROR=${MIRROR:-https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp}
+
+# If user defined the OPENSHIFT_VERSION environment variable then use it.
+if test -n "${OPENSHIFT_VERSION-}"; then
+    OPENSHIFT_RELEASE_VERSION=${OPENSHIFT_VERSION}
+    echo "Using release ${OPENSHIFT_RELEASE_VERSION} from OPENSHIFT_VERSION"
```

Comment:
Might need some check that it's 4.11.x, and not set to `OPENSHIFT_VERSION=4.9.13` for example?

Reply:
@cfergeau yes it still need some improvement in term of mismatch of openshift version and the dist-git branch we use.